### ISSUE TITLE
Newznab Module - Bad SSL Handshake

### ISSUE THREAD

**Issue by MiTeePingWin** [Hellowlol/HTPC-Manager#373]
I just updated to the October 27th master2 branch.
Thew Newznab module no longer works with either DOGNzb or NZBGeek (SSL) but it works fine with 6boxme. I just used this module yesterday with no issues, before the update.

The error I get is:
ERROR   Failed to fetch results [Errno bad handshake] [('SSL routines', 'SSL23_GET_SERVER_HELLO', None)]

Is this an error on my end with openssl (Windows)?

**Comment by Hellowlol** [comment]
Hi!

I would like to have some more info:

What ssl version is installed?
Full log please.
Screenshot of the about page

I believe the issue is on your end yes

**Comment by Hellowlol** [comment]
nzbgeek works fine here. dog doesnt not but thats because the api is temporary disabled.

**Comment by MiTeePingWin** [comment]
Log: http://pastebin.com/3959Z48S

I'm using Python2.7 with pyOpenSSL 0.15.1
CP and Sonarr work just fine.

**Comment by Hellowlol** [comment]
What minor version? dog works fine here btw.

**Comment by MiTeePingWin** [comment]
I was running 12.1 but I updated last night to the 15.1 .whl package from here:
https://pypi.python.org/pypi/pyOpenSSL

I still can't get Dog or NZBGeek to work with the module.
NZBMegaSearch works just fine, though...?

**Comment by Hellowlol** [comment]
Sorry, what minor python version are you using? Dunno wtfs going on. Have you installed requests via pip?

**Comment by MiTeePingWin** [comment]
Right, sorry. Python 2.7.6
I'm going to dump and re-clone and try again.

**Comment by Hellowlol** [comment]
I don't think that will solve anything. can you paste the result of pip list ?

**Comment by MiTeePingWin** [comment]
It didn't, sadly. I re-ran with debug logs, fresh install without any settings.
New log: http://pastebin.com/uRtsNZZ5
I was missing requests. I installed, but still no luck, same error.

C:\Python27\Scripts>pip list
cffi (1.3.0)
cryptography (1.0.2)
enum34 (1.0.4)
idna (2.0)
ipaddress (1.0.14)
PIL (1.1.7)
pip (1.5.6)
psutil (3.2.2)
pyasn1 (0.1.9)
pycparser (2.14)
pycurl (7.19.0.2)
pyOpenSSL (0.15.1)
pywin32 (219)
requests (2.8.1)
setuptools (7.0)
six (1.10.0)

**Comment by Hellowlol** [comment]
Ok, can you try to run https://gist.github.com/Hellowlol/8df98532ed6c43d27bea manually?. Remember to edit the apikey.

**Comment by MiTeePingWin** [comment]
(I had to edit line 1 from 'request' to 'requests')
So now I'm thoroughly confused:
I ran it once and I was denied. I logged into DOG, checked I was still valid, checked the API key etc. but made no changes.
I tried the script again and it worked. I tried the newznab module again and now Dog WORKS.

NZBGeek still doesn't work though. 
I changed your test script to:url = 'https://api.nzbgeek.info/api?o=xml&apikey=%s&t=search&q=Mission&extended=1' % apikey
The script works and I see results from NZBGeek, however I'm still getting the error in the newznab module even though the return request is valid per the logs:

2015-10-28 09:04:17 modules.newznab ERROR   Failed to fetch results [Errno bad handshake] [('SSL routines', 'SSL23_GET_SERVER_HELLO', None)]
2015-10-28 09:04:16 modules.newznab DEBUG   Fetching search results from https://api.nzbgeek.info/api?o=xml&apikey=********************************&t=search&q=Quantico&extended=1
2015-10-28 09:04:16 modules.newznab DEBUG   Searching for Quantico category on indexer NZBGeek

**Comment by Hellowlol** [comment]
Sorry about the typo. Hmm, very strange... So just to be clear, nzbgeek works with the script and dog works in htpc-manager now?

**Comment by MiTeePingWin** [comment]
THANK YOU! This fixed it.
I downgraded to pyOpenSSL 0.13.1 (from 0.15.1) and upgraded to Windows OpenSSL 1.0.2d. To do this I had to replace the openssl.exe file located within C:\Program Files (x86)\Git\bin  as it had an old version. ('Which openssl' was returning this old version and not 1.0.2d) Not sure if this was necessary or not.

This fixed my issue.
It seems like the problem with Dog was that I haven't browsed to the site recently, thus the cert wasn't recognized (?)

You're right on the SSL cert for Geek, I would have never figured that out.
Thank you again!!
